pkg/ccl/logictestccl/testdata/logic_test/fk_read_committed

```diff
@@ -98,3 +98,81 @@ INSERT INTO e VALUES (2)
 
 statement ok
 UPDATE a SET a = 2 WHERE a = 1
+
+subtest fk_cascade_race_150282
+
+statement ok
+CREATE TABLE parent_150282 (
+  p INT PRIMARY KEY,
+  i INT,
+  j INT,
+  INDEX (i),
+  INDEX (j)
+);
+
+statement ok
+CREATE TABLE child_150282 (
+  c INT PRIMARY KEY,
+  p INT REFERENCES parent_150282 (p) ON DELETE CASCADE ON UPDATE CASCADE,
+  INDEX (p)
+);
+
+statement ok
+INSERT INTO parent_150282 VALUES (1, 2, 3);
+
+statement ok
+BEGIN ISOLATION LEVEL READ COMMITTED;
+
+statement ok
+SELECT 1;
+
+statement async fk_delete
+WITH sleep AS (SELECT pg_sleep(1)) DELETE FROM parent_150282@parent_150282_i_idx WHERE i = 2;
```

Comment:
nit: I think we should have a third test for the on-delete-cascade fast-path, since that's a separate code path that you updated. For that to occur I think we'd need a filter on the `DELETE` of `p = 2`, which can be directly translated into a filter on the child table, eliminating a scan of the buffer of deleted parent rows, e.g.: https://github.com/cockroachdb/cockroach/blob/ce98c91e9aea368d7c2a15dfd096d1758c2e6398/pkg/sql/opt/optbuilder/testdata/fk-on-delete-cascade#L17-L44

I don't think this should block this backport, or that we necessarily need to backport such a test. But it would be good to have on `master`.